### PR TITLE
AVM: Show opcode context for logicsigs (not just apps)

### PR DESCRIPTION
**Pull request by jannotti**
Users find the extra info about the previous opcodes to be a helpful addition when debugging an app failure.  This extends the support to LogicSigs.

### CODE REVIEW

File: data/transactions/logic/eval.go [hunk 1]
```diff
@@ -5522,8 +5551,8 @@ func opBlock(cx *EvalContext) error {
 	}
 }
 
-// PcDetails return PC and disassembled instructions at PC up to 2 opcodes back
-func (cx *EvalContext) PcDetails() (pc int, dis string) {
+// pcDetails return PC and disassembled instructions at PC up to 2 opcodes back
+func (cx *EvalContext) pcDetails() (pc int, dis string) {
```
Comment (bbroder-algo) [inline data/transactions/logic/eval.go:5555]: Can we use this opportunity to replace newlines in the string with semicolons

Comment (jannotti) [reply data/transactions/logic/eval.go:5555]: yeah, nice idea, since they are legal TEAL now.

File: data/transactions/logic/eval.go [hunk 2]
```diff
@@ -678,33 +678,46 @@ func (st StackType) Typed() bool {
...
 
-func (e ClearStateBudgetError) Error() string {
-	return fmt.Sprintf("Attempted ClearState execution with low OpcodeBudget %d", e.offered)
+// Error satisfies builtin interface `error`
+func (err EvalError) Error() string {
+	var msg string
+	if err.logicsig {
+		msg = fmt.Sprintf("rejected by logic err=%v", err.Err)
+	} else {
+		msg = fmt.Sprintf("logic eval error: %v", err.Err)
```
Comment (bbroder-algo) [inline data/transactions/logic/eval.go:708]: tricky distinction here between "logic eval error" and "logicsig logic error"

Comment (jannotti) [reply data/transactions/logic/eval.go:708]: I don't like that language necessarily, I'm just matching the existing wording to decrease test churn.

willing to change.

**Review by bbroder-algo — APPROVED** [review]

**Review by jasonpaulos — APPROVED** [review]

**Pull request merged**